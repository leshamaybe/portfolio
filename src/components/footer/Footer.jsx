import React from 'react';
import Social from '../social/Social';
import Link from 'next/link';
import { usePathname } from 'next/navigation';

const Footer = () => {
    const pathname = usePathname();

    return (
        <footer className="max-w-[1528px] w-full flex flex-col py-10 px-4">
            {pathname == '/' && (
                <div className="flex flex-col items-center text-primary-color mb-10+10">
                    <h1 className="uppercase font-monument text-20+25 text-center font-extrabold mb-10+10">
                        Have a project in mind or want to talk?
                    </h1>
                    <span className="text-18+7">
                        Say hey at
                        <Link
                            className="py-[5px] pr-[5px] ml-[5px] text-second-color bg-primary-black bg-wave hover:bg-hover-wave bg-repeat-x bg-[0_100%] bg-[length:auto_5px] "
                            href={'mailto:vvaleksey@icloud.com'}>
                            {' '}
                            vvaleksey@icloud.com
                        </Link>
                    </span>
                </div>
            )}
            <div className="flex items-center justify-between w-full">
                <h2 className='text-12+3'>Â© 2023 leshamaybe</h2>
                <Social />
            </div>
        </footer>
    );
};

export default Footer;
