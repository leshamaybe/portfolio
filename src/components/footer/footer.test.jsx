import Footer from './Footer';
import { render } from '@testing-library/react';

describe('Footer', () => {
    it('Footer renders correctly', () => {
        const { container } = render(<Footer />);
        expect(container).toMatchSnapshot();
    });
});