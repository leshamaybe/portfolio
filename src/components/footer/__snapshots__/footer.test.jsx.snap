// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Footer renders correctly 1`] = `
<div>
  <footer
    class="footer"
  >
    <div
      class="container"
    >
      <div
        class="bottom"
      >
        <h2>
          Â© 2023 leshamaybe
        </h2>
        <div
          class="social"
        >
          <a
            href="https://github.com/leshamaybe"
          >
            <svg
              fill="grey"
              height="24px"
              viewBox="0 0 50 50"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <defs>
                <lineargradient
                  gradientTransform="rotate(95)"
                  id="Gradient-1"
                  x1="2.49%"
                  x2="97.19%"
                  y1="0"
                  y2="0"
                >
                  <stop
                    offset="0.2"
                    stop-color="#00baff"
                  />
                  <stop
                    offset="0.97"
                    stop-color="#063ef9"
                  />
                </lineargradient>
              </defs>
              <path
                d="M17.791,46.836C18.502,46.53,19,45.823,19,45v-5.4c0-0.197,0.016-0.402,0.041-0.61C19.027,38.994,19.014,38.997,19,39 c0,0-3,0-3.6,0c-1.5,0-2.8-0.6-3.4-1.8c-0.7-1.3-1-3.5-2.8-4.7C8.9,32.3,9.1,32,9.7,32c0.6,0.1,1.9,0.9,2.7,2c0.9,1.1,1.8,2,3.4,2 c2.487,0,3.82-0.125,4.622-0.555C21.356,34.056,22.649,33,24,33v-0.025c-5.668-0.182-9.289-2.066-10.975-4.975 c-3.665,0.042-6.856,0.405-8.677,0.707c-0.058-0.327-0.108-0.656-0.151-0.987c1.797-0.296,4.843-0.647,8.345-0.714 c-0.112-0.276-0.209-0.559-0.291-0.849c-3.511-0.178-6.541-0.039-8.187,0.097c-0.02-0.332-0.047-0.663-0.051-0.999 c1.649-0.135,4.597-0.27,8.018-0.111c-0.079-0.5-0.13-1.011-0.13-1.543c0-1.7,0.6-3.5,1.7-5c-0.5-1.7-1.2-5.3,0.2-6.6 c2.7,0,4.6,1.3,5.5,2.1C21,13.4,22.9,13,25,13s4,0.4,5.6,1.1c0.9-0.8,2.8-2.1,5.5-2.1c1.5,1.4,0.7,5,0.2,6.6c1.1,1.5,1.7,3.2,1.6,5 c0,0.484-0.045,0.951-0.11,1.409c3.499-0.172,6.527-0.034,8.204,0.102c-0.002,0.337-0.033,0.666-0.051,0.999 c-1.671-0.138-4.775-0.28-8.359-0.089c-0.089,0.336-0.197,0.663-0.325,0.98c3.546,0.046,6.665,0.389,8.548,0.689 c-0.043,0.332-0.093,0.661-0.151,0.987c-1.912-0.306-5.171-0.664-8.879-0.682C35.112,30.873,31.557,32.75,26,32.969V33 c2.6,0,5,3.9,5,6.6V45c0,0.823,0.498,1.53,1.209,1.836C41.37,43.804,48,35.164,48,25C48,12.318,37.683,2,25,2S2,12.318,2,25 C2,35.164,8.63,43.804,17.791,46.836z"
              />
            </svg>
          </a>
          <a
            href="https://vk.com/leshamaybe"
          >
            <svg
              fill="grey"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="st0"
                d="M13.162 18.994c.609 0 .858-.406.851-.915-.031-1.917.714-2.949 2.059-1.604 1.488 1.488 1.796 2.519 3.603 2.519h3.2c.808 0 1.126-.26 1.126-.668 0-.863-1.421-2.386-2.625-3.504-1.686-1.565-1.765-1.602-.313-3.486 1.801-2.339 4.157-5.336 2.073-5.336h-3.981c-.772 0-.828.435-1.103 1.083-.995 2.347-2.886 5.387-3.604 4.922-.751-.485-.407-2.406-.35-5.261.015-.754.011-1.271-1.141-1.539-.629-.145-1.241-.205-1.809-.205-2.273 0-3.841.953-2.95 1.119 1.571.293 1.42 3.692 1.054 5.16-.638 2.556-3.036-2.024-4.035-4.305-.241-.548-.315-.974-1.175-.974h-3.255c-.492 0-.787.16-.787.516 0 .602 2.96 6.72 5.786 9.77 2.756 2.975 5.48 2.708 7.376 2.708z"
              />
            </svg>
          </a>
          <a
            href="https://t.me/blue_rari"
          >
            <svg
              fill="grey"
              height="24px"
              viewBox="0 0 48 48"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.83,23.616c12.568-5.529,28.832-12.27,31.077-13.203c5.889-2.442,7.696-1.974,6.795,3.434 c-0.647,3.887-2.514,16.756-4.002,24.766c-0.883,4.75-2.864,5.313-5.979,3.258c-1.498-0.989-9.059-5.989-10.7-7.163 c-1.498-1.07-3.564-2.357-0.973-4.892c0.922-0.903,6.966-6.674,11.675-11.166c0.617-0.59-0.158-1.559-0.87-1.086 c-6.347,4.209-15.147,10.051-16.267,10.812c-1.692,1.149-3.317,1.676-6.234,0.838c-2.204-0.633-4.357-1.388-5.195-1.676 C1.93,26.43,2.696,24.995,5.83,23.616z"
              />
            </svg>
          </a>
          <a
            href="/"
          >
            <svg
              fill="grey"
              height="24px"
              viewBox="0 0 51 51"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M 8 3.0097656 C 4.53 3.0097656 2.0097656 5.0892187 2.0097656 7.9492188 C 2.0097656 10.819219 4.59 12.990234 8 12.990234 C 11.47 12.990234 13.990234 10.870625 13.990234 7.890625 C 13.830234 5.020625 11.36 3.0097656 8 3.0097656 z M 3 15 C 2.45 15 2 15.45 2 16 L 2 45 C 2 45.55 2.45 46 3 46 L 13 46 C 13.55 46 14 45.55 14 45 L 14 16 C 14 15.45 13.55 15 13 15 L 3 15 z M 18 15 C 17.45 15 17 15.45 17 16 L 17 45 C 17 45.55 17.45 46 18 46 L 27 46 C 27.552 46 28 45.552 28 45 L 28 30 L 28 29.75 L 28 29.5 C 28 27.13 29.820625 25.199531 32.140625 25.019531 C 32.260625 24.999531 32.38 25 32.5 25 C 32.62 25 32.739375 24.999531 32.859375 25.019531 C 35.179375 25.199531 37 27.13 37 29.5 L 37 45 C 37 45.552 37.448 46 38 46 L 47 46 C 47.55 46 48 45.55 48 45 L 48 28 C 48 21.53 44.529063 15 36.789062 15 C 33.269062 15 30.61 16.360234 29 17.490234 L 29 16 C 29 15.45 28.55 15 28 15 L 18 15 z"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
